src/compiler/call.ts

```diff
@@ -479,11 +481,11 @@ export function getPropertyAccessExpressionType(
 	const subExpType = subExp.getType();
 	const property = expression.getName();
 
-	if (isArrayMethodType(expType)) {
+	if (isArrayMethodType(expType) || (isArrayType(subExpType) && property === "length")) {
```

Comment:
this probably isn't the best way to do this

Reply:
It isn't. I already have a fix for this in my PR https://github.com/roblox-ts/roblox-ts/pull/340. Please remove this.